gbpservice/nfp/config_orchestrator/agent/loadbalancerv2.py

```diff
@@ -0,0 +1,457 @@
+                                  'eventdata': request_data}
+        ev = self._sc.new_event(id=event_id,
+                                key=event_id, data=event_data)
+        self._sc.post_event(ev)
+
+    def update_status(self, context, notification_data):
+        notification = notification_data['notification'][0]
+        notification_info = notification_data['info']
+        resource_data = notification['data']
+        obj_type = resource_data['obj_type']
+        obj_id = resource_data['obj_id']
+        service_type = notification_info['service_type']
+
+        rpcClient = transport.RPCClient(a_topics.LBV2_NFP_PLUGIN_TOPIC)
+        rpcClient.cctxt = rpcClient.client.prepare(
+            version=const.LOADBALANCERV2_RPC_API_VERSION)
+
+        lb_p_status = const.ACTIVE
+        lb_o_status = None
+        obj_p_status = resource_data['provisioning_status']
+        obj_o_status = resource_data['operating_status']
+
+        msg = ("NCO received LB's update_status API, making an update_status "
+               "RPC call to plugin for %s: %s with status %s" % (
+                   obj_type, obj_id, obj_p_status))
+        LOG(LOGGER, 'INFO', '%s' % (msg))
+
+        if obj_type == 'healthmonitor':
+                obj_o_status = None
+
+        if obj_type != 'loadbalancer':
+            rpcClient.cctxt.cast(context, 'update_status',
+                                 obj_type=obj_type,
+                                 obj_id=obj_id,
+                                 provisioning_status=obj_p_status,
+                                 operating_status=obj_o_status)
+        else:
+            lb_o_status = const.ONLINE
+            if obj_p_status == const.ERROR:
+                lb_p_status = const.ERROR
+                lb_o_status = const.OFFLINE
+
+        rpcClient.cctxt.cast(context, 'update_status',
+                             obj_type='loadbalancer',
+                             obj_id=resource_data['root_lb_id'],
+                             provisioning_status=lb_p_status,
+                             operating_status=lb_o_status)
+
+        if obj_type.lower() == 'loadbalancer':
+            nf_id = notification_info['context']['network_function_id']
+            lb_id = notification_info['context']['loadbalancer_id']
+            # sending notification to visibility
+            event_data = {'context': context.to_dict(),
+                          'nf_id': nf_id,
+                          'loadbalancer_id': lb_id,
+                          'service_type': service_type,
+                          'resource_id': lb_id
+                          }
+            ev = self._sc.new_event(id='SERVICE_CREATE_PENDING',
+                                    key='SERVICE_CREATE_PENDING',
+                                    data=event_data, max_times=24)
+            self._sc.poll_event(ev)
+
+    # TODO(jiahao): copy from v1 agent need to review
+    # def update_pool_stats(self, context, notification_data):
```

Comment:
We need to check the LBaaSV2 resources and check where pool stats are shown. Need to add this. This is invoked in the configurator driver but not handled here.